### PR TITLE
- Listing expiry time counter

### PR DESCRIPTION
**Pull request by cis-shubham-t**
- Added the listing expiry time in listing preview.
- Added the listing expiry count-down in listing preview.
- Added the listing expiry time in the checkout process.

### CODE REVIEW

**Comment by cis-shubham-t** [comment]
Task: https://particl.atlassian.net/projects/PBD/issues/PBD-95

**Comment by cis-shubham-t** [comment]
hey @AllienWorks 
I've added the time counter for the listing could you please have a look at the listing preview popup and make highlighted (maybe with red).

I'm not put the condition for the counter only visible if the listing expiry in the next 24 hours.  so you can see the counter time for the each listing.

**Comment by AllienWorks** [comment]
I've styled this, both on listing preview & in cart. Also activated the condition for hiding of the "expires soon" messages (@cis-shubham-t please double-check if it's working correctly :)

screenshot from 2018-12-21 17-45-04

File: src/app/market/listings/preview-listing/preview-listing.component.ts [hunk 1]
```diff
@@ -54,6 +54,7 @@ export class PreviewListingComponent implements OnInit, OnDestroy {
     if (this.data.listing) {
       this.images = this.data.listing.imageCollection.imageUrls;
     }
+    console.log('this.data', this.data);
```
Comment (zaSmilingIdiot) [inline src/app/market/listings/preview-listing/preview-listing.component.ts]: Please remove this line, or change it to a debug log statement.

File: src/app/market/listings/preview-listing/preview-listing.component.html [hunk 2]
```diff
@@ -52,8 +51,17 @@ <h1>{{ data?.listing?.title }}</h1>
...
-
+      
       <table class="meta">
+        <tr class="expires-soon" *ngIf="data?.listing?.isAboutToExpire">
+          <th class="title">
+            <mat-icon fontSet="partIcon" fontIcon="part-circle-alert"></mat-icon>
+            Expires soon
+          </th>
+          <td class="value">
+            {{ data?.listing?.expireIn }} left
```
Comment (zaSmilingIdiot) [inline src/app/market/listings/preview-listing/preview-listing.component.html]: Maybe take the word 'left' out here... if the time is counting down then the meaning is somewhat clear.

File: src/app/core/market/api/template/template.model.ts [hunk 3]
```diff
@@ -180,6 +183,19 @@ export class Template {
     'Expires ' + new DateFormatter(new Date(this.object.ListingItems[0].expiredAt)).dateFormatter(false).substr(0, 16) : '';
   }
 
+  get isAboutToExpire(): any {
```
Comment (zaSmilingIdiot) [inline src/app/core/market/api/template/template.model.ts]: The body of this function can be collapsed/shortened into a single line:
`return this.object.expiredAt > +new Date() && ((this.object.expiredAt - +new Date()) <= 86400000);`

Comment (zaSmilingIdiot) [reply src/app/core/market/api/template/template.model.ts]: Also, the return type is boolean, not any

File: src/app/core/market/api/template/template.model.ts [hunk 4]
```diff
@@ -180,6 +183,19 @@ export class Template {
...
+
+    // 86400000 m seconds in one day.
+    if (this.object.expiredAt > +new Date() && ((this.object.expiredAt - +new Date()) <= 86400000)) {
+      return true;
+    }
+    return false;
+  }
+
+  get expireIn(): any {
```
Comment (zaSmilingIdiot) [inline src/app/core/market/api/template/template.model.ts]: please change the return type

**Review by zaSmilingIdiot — CHANGES_REQUESTED** [review]

**Review by zaSmilingIdiot — APPROVED** [review]

**Pull request merged**